File: src/test/java/no/nav/arbeidsgiver/min_side/controller/RestResponseEntityExceptionHandlerTest.java
package no.nav.arbeidsgiver.min_side.controller;

import no.nav.security.token.support.spring.test.EnableMockOAuth2Server;
import org.junit.jupiter.api.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;
import org.springframework.web.context.request.WebRequest;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.mockito.Mockito.mock;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@RunWith(SpringRunner.class)
@EnableMockOAuth2Server
class RestResponseEntityExceptionHandlerTest {
    @Autowired
    RestResponseEntityExceptionHandler restResponseEntityExceptionHandler;

    @Test
    void loggerMedSirkulærReferanseFraInit() {
        Exception circular = new Exception("foo");
        Exception e2 = new Exception(circular);
        circular.initCause(e2);

        assertDoesNotThrow(() -> {
            restResponseEntityExceptionHandler.handleInternalError(
                    new RuntimeException("wat", circular),
                    mock(WebRequest.class)
            );
        });
    }

    @Test
    void loggerMedSirkulærReferanseFraSupressed() {
        Exception circular = new Exception("foo");
        Exception e2 = new Exception(circular);
        circular.addSuppressed(e2);

        assertDoesNotThrow(() -> {
            restResponseEntityExceptionHandler.handleInternalError(
                    new RuntimeException("wat", circular),
                    mock(WebRequest.class)
            );
        });
    }
}

File: src/test/kotlin/no/nav/arbeidsgiver/min_side/controller/RestResponseEntityExceptionHandlerTest.kt
package no.nav.arbeidsgiver.min_side.controller

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.mockito.Mockito
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.mock.mockito.MockBean
import org.springframework.security.oauth2.jwt.JwtDecoder
import org.springframework.web.context.request.WebRequest

@MockBean(JwtDecoder::class)
@SpringBootTest(
    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
    properties = [
        "spring.flyway.enabled=false"
    ],
)
class RestResponseEntityExceptionHandlerTest {

    @Autowired
    lateinit var restResponseEntityExceptionHandler: RestResponseEntityExceptionHandler

    @Test
    fun loggerMedSirkulærReferanseFraInit() {
        val circular = Exception("foo")
        val e2 = Exception(circular)
        circular.initCause(e2)
        Assertions.assertDoesNotThrow {
            restResponseEntityExceptionHandler.handleInternalError(
                RuntimeException("wat", circular),
                Mockito.mock(WebRequest::class.java)
            )
        }
    }

    @Test
    fun loggerMedSirkulærReferanseFraSupressed() {
        val circular = Exception("foo")
        val e2 = Exception(circular)
        circular.addSuppressed(e2)
        Assertions.assertDoesNotThrow {
            restResponseEntityExceptionHandler.handleInternalError(
                RuntimeException("wat", circular),
                Mockito.mock(WebRequest::class.java)
            )
        }
    }
}